src/test/scheduler_tests.cpp

```diff
@@ -56,10 +54,10 @@ BOOST_AUTO_TEST_CASE(manythreads)
 
     boost::mutex counterMutex[10];
     int counter[10] = { 0 };
-    boost::random::mt19937 rng(42);
-    boost::random::uniform_int_distribution<> zeroToNine(0, 9);
-    boost::random::uniform_int_distribution<> randomMsec(-11, 1000);
-    boost::random::uniform_int_distribution<> randomDelta(-1000, 1000);
+    FastRandomContext rng(42);
+    auto zeroToNine = [](FastRandomContext& rc) { return rc.randrange(10); }; // [0, 9]
+    auto randomMsec = [](FastRandomContext& rc) { return -11 + rc.randrange(1012); }; // [-11, 1000]
+    auto randomDelta = [](FastRandomContext& rc) { return -1000 + rc.randrange(2001); }; // [-1000, 1000]
```

Comment:
Good catch, hadn't realized `randrange` returns an unsigned value. This is a risk with `auto`.